cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -157,10 +203,9 @@ public void run() {
     }
   }
 
-  // At least two computing thread is needed if precomputing is disabled. One thread for submitting and waiting for
-  // the proposal computing to finish, another one for compute the proposals.
+  // If precomputing is disabled, need one thread to compute the proposals.
   private int numProposalComputingThreads() {
-    return _numPrecomputingThreads > 0 ? _numPrecomputingThreads : 2;
+    return _numPrecomputingThreads > 0 ? _numPrecomputingThreads : 1;
```

Comment:
This is a special case for handling if precomputing is disabled: At least two computing thread is needed if precomputing is disabled. One thread for submitting and waiting for the proposal computing to finish, another one for compute the proposals.

Let's roll this function back to the initial version of it. See https://github.com/linkedin/cruise-control/commit/451b378f6d0b6de44cfc7588724a8e83381dbbaf for the relevant patch that adds this fix.

Can we also add a test that checks if `numProposalComputingThreads()` returns `2` if `_numPrecomputingThreads` is `0`?

Reply:
Hi Efe, this actually confuses me.
looks to me, the thread for submitting and waiting for the proposal computing to finish is the main thread. so if _numPrecomputingThreads is 0, one more thread is enough?

 from code, where _proposalPrecomputingExecutor.submit is used are in GoalOptimizer.java

line 215
    for (int i = 0; i < (_numPrecomputingThreads > 0 ? _numPrecomputingThreads : 1); i++) {
      futures.add(_proposalPrecomputingExecutor.submit(new ProposalCandidateComputer(_goalByPriorityForPrecomputing.get(i))));
    }

and line 308
              if (numWaitingThread == 0) {
                // No precomputing thread is available, schedule a computing and wait for the cache update.
                _proposalPrecomputingExecutor.submit(this::computeBestProposal);
              }

Reply:
reverted the change.